lib/auth/auth.go

```diff
@@ -4289,6 +4312,9 @@ func (a *Server) UpdateKubernetesCluster(ctx context.Context, kubeCluster types.
 
 // DeleteKubernetesCluster deletes a kubernetes cluster resource.
 func (a *Server) DeleteKubernetesCluster(ctx context.Context, name string) error {
+	if err := enforceLicense(types.KindKubernetesCluster); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I wonder if its necessary to enforce license when deleting a kube cluster resource? I can only imagine that this would happen if the user has kube clusters already registered, yet their Teleport cluster is no longer licensed to use kubernetes?

If that scenario did happen, wouldn't it be better to allow them delete the resources they can't use and maybe just log this as a warning?